### PR TITLE
Cancer Type and Mutation Count columns in ResultsViewMutationTable

### PR DESCRIPTION
**Pull request by adamabeshouse**
Also a slightly different system for deciding which columns to exclude.

# Checks
- [x] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [x] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

File: src/pages/resultsView/mutation/ResultsViewMutationTable.tsx [hunk 1]
```diff
@@ -40,19 +48,62 @@ export default class ResultsViewMutationTable extends MutationTable<IResultsView
             MutationTableColumnType.MUTATION_ASSESSOR,
             MutationTableColumnType.COSMIC,
             MutationTableColumnType.TUMOR_ALLELE_FREQ,
-            MutationTableColumnType.NORMAL_ALLELE_FREQ
+            MutationTableColumnType.NORMAL_ALLELE_FREQ,
+            MutationTableColumnType.CANCER_TYPE,
+            MutationTableColumnType.NUM_MUTATIONS
```
Comment (onursumer) [inline src/pages/resultsView/mutation/ResultsViewMutationTable.tsx]: These new columns (`CancerType` and `NumMutations`) can be moved to the base class if we like. Seems like they don't have special dependencies to any other component. But we can leave them here too if we think we don't have any general use case for these columns.

File: src/shared/components/mutationTable/CancerTypeCache.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import ClinicalDataCache from "../../lib/ClinicalDataCache";
+
+export default class CancerTypeCache extends ClinicalDataCache {
+    constructor(studyId:string) {
+        super(studyId, "CANCER_TYPE", "SAMPLE", "SUMMARY");
+    }
+}
```
Comment (onursumer) [inline src/shared/components/mutationTable/CancerTypeCache.ts]: Should we move this file to `src/shared/cache` for consistency?

File: src/shared/components/mutationTable/MutationCountCache.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    return client.fetchMutationCountsInGeneticProfileUsingPOST({
+        geneticProfileId,
+        sampleIds
+    });
+}
+export default class MutationCountCache extends LazyMobXCache<MutationCount, string> {
+    constructor(geneticProfileId:string) {
+        super(q=>q, (d:MutationCount)=>d.sampleId, fetch, geneticProfileId);
+    }
+}
```
Comment (onursumer) [inline src/shared/components/mutationTable/MutationCountCache.ts]: Should we move this file to `src/shared/cache` for consistency?

File: src/shared/components/mutationTable/MutationTable.tsx [hunk 4]
```diff
@@ -77,10 +77,12 @@ export enum MutationTableColumnType {
...
     VAR_READS_N,
     REF_READS,
-    VAR_READS
+    VAR_READS,
+    CANCER_TYPE,
+    NUM_MUTATIONS
 }
 
-type MutationTableColumn = Column<Mutation[]>&{order?:number};
+type MutationTableColumn = Column<Mutation[]>&{order?:number, shouldExclude?:()=>boolean};
```
Comment (onursumer) [inline src/shared/components/mutationTable/MutationTable.tsx]: good idea!

File: src/shared/lib/ClinicalDataCache.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                attributeId:string,
+                clinicalDataType:"SAMPLE"|"PATIENT",
+                projection:"ID" | "SUMMARY" | "DETAILED" | "META") {
+        super(
+            q=>q.studyId+"~"+q.entityId,
+            d=>studyId+"~"+d.entityId,
+            fetch,
+            attributeId, clinicalDataType, projection);
+    }
+}
```
Comment (onursumer) [inline src/shared/lib/ClinicalDataCache.ts]: move to `src/shared/cache` instead?

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts [hunk 6]
```diff
@@ -35,6 +35,8 @@ import {IMutSigData} from "shared/model/MutSig";
 import {ClinicalInformationData} from "shared/model/ClinicalInformation";
 import VariantCountCache from "shared/cache/VariantCountCache";
 import CopyNumberCountCache from "./CopyNumberCountCache";
+import CancerTypeCache from "../../../shared/components/mutationTable/CancerTypeCache";
+import MutationCountCache from "../../../shared/components/mutationTable/MutationCountCache";
```
Comment (onursumer) [inline src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: I like `shared/components/` more than `../../../shared/components` :)

File: src/shared/components/mutationTable/CancerTypeCache.ts (hunk 2 above)
Comment (adamabeshouse) [inline src/shared/components/mutationTable/CancerTypeCache.ts]: done

File: src/shared/components/mutationTable/MutationCountCache.ts (hunk 3 above)
Comment (adamabeshouse) [inline src/shared/components/mutationTable/MutationCountCache.ts]: done

File: src/shared/lib/ClinicalDataCache.ts (hunk 5 above)
Comment (adamabeshouse) [inline src/shared/lib/ClinicalDataCache.ts]: done

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts (hunk 6 above)
Comment (adamabeshouse) [inline src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: done

File: src/pages/resultsView/mutation/ResultsViewMutationTable.tsx (hunk 1 above)
Comment (adamabeshouse) [inline src/pages/resultsView/mutation/ResultsViewMutationTable.tsx]: good idea

**Comment by adamabeshouse** [comment]
[(5) Results Page: Mutations Tab: Table: Cancer Type column](https://trello.com/c/dv1FFmBp/235-5-results-page-mutations-tab-table-cancer-type-column)

**Comment by adamabeshouse** [comment]
[(5) Results page: Mutations Tab: Table: #Muts column](https://trello.com/c/ZrQnkmb9/236-5-results-page-mutations-tab-table-muts-column)

**Pull request merged**